api/query/atoms.go

```diff
@@ -257,7 +258,14 @@ func (t AbstractTriaged) BindToRuns(runs ...shared.TestRun) ConcreteQuery {
 	}
 	for _, run := range runs {
 		if t.Product == nil || t.Product.Matches(run) {
-			metadata, _ := shared.GetMetadataResponse(runs, logrus.StandardLogger(), t.metadataFetcher)
+			includeTestLevel := false
+			metadataRuns := runs
+			// Product being nil means that we want test-level issues.
+			if (t.Product == nil) {
+				includeTestLevel = true
+				metadataRuns = make([]shared.TestRun, 0)
```

Comment:
```suggestion
```
We don't need this anymore as test-level metadata will [break before the loop](https://github.com/web-platform-tests/wpt.fyi/pull/2440/files#diff-48d7909c1ce4d6c5e32eb0144ec673308fee32ab1f56f6b265010b988934e5ebR144)

Reply:
I think we still need this; that break is only for a single result, for a single link in the metadata. For example, if you had:

```
links:
    - url: http://example.org
      results:
        - test: foo.html
    - product: chrome
      url: https://crbug.com/1234
      results:
        - test: bar.html
```

And you searched `triaged:test-issue`, the loop in `constructMetadataResponse` (assuming includeTestLevel=true and runs = [chrome, firefox, safari]) is:

```
1. for i := range data.Links
    a. link := data.Links[0] = (url: http://example.org, product: nil, results: [ test: foo.html ])
    b. for result := range link.Results
        i. result = (test: foo.html)
        ii. link.Product.BrowserName == "", and includeTestLevel=true, so add link and break
    [ next iteration of outer loop ]
    a. link := data.Links[1] = (url: http://crbug.com/1234, product: Chrome, results: [ test: bar.html ])
    b. for result := range link.Results
        i. result = (test: bar.html)
        ii. link.Product.BrowserName != ""
        iii. for productSpec := range [chrome, firefox, safari]
            1. productSpec = chrome
            2. if link.Product matches chrome, success, so **add link and break**
```

So we end up adding bar.html even though we only wanted test-level issues!

Reply:
Yea you are right. I was confused a little bit. Now we can return 

- all metadata by passing in `runs` and true for `includeTestLevel`
- metadata without test-level by passing in `runs` and false for `includeTestLevel`
- test-level only by passing empty runs and true for `includeTestLevel`